chia/consensus/cost_calculator.py

```diff
@@ -2,32 +2,28 @@
 from typing import List, Optional
 
 from chia.consensus.condition_costs import ConditionCost
-from chia.full_node.mempool_check_conditions import get_name_puzzle_conditions
 from chia.types.blockchain_format.program import SerializedProgram
 from chia.types.condition_opcodes import ConditionOpcode
 from chia.types.name_puzzle_condition import NPC
-from chia.util.ints import uint16, uint64
+from chia.util.ints import uint64, uint16
 from chia.util.streamable import Streamable, streamable
 
 
 @dataclass(frozen=True)
 @streamable
-class CostResult(Streamable):
+class NPCResult(Streamable):
     error: Optional[uint16]
     npc_list: List[NPC]
-    cost: uint64
+    clvm_cost: uint64  # CLVM cost only, cost of conditions and tx size is not included
 
 
-def calculate_cost_of_program(program: SerializedProgram, cost_per_byte: int, strict_mode: bool = False) -> CostResult:
+def calculate_cost_of_program(program: SerializedProgram, npc_result: NPCResult, cost_per_byte: int) -> uint64:
```

Comment:
the `program` and `cost_per_byte` parameters here seem a bit unnecessary. all we do is to multiply the size of the program by `cost_per_byte`.

Ideally, this would happen *first*, before we even execute the CLVM code, to allow setting the max cost for the CLVM program to `MAX_BLOCK_COST_CLVM` - `len(program) * COST_PER_BYTE`.

It seems like this would belong somewhere in the `npc_result`

Reply:
this is not a big deal though, we can simplify this later